lib/ForkProcessSandbox.cpp

```diff
@@ -0,0 +1,48 @@
+
+#include "ForkProcessSandbox.h"
+
+#include "TestResult.h"
+
+#include <sys/mman.h>
+#include <unistd.h>
+
+Mutang::ExecutionResult Mutang::ForkProcessSandbox::run(std::function<void (ExecutionResult *)> Func) {
+
+  void *SharedMemory = mmap(NULL,
+                            sizeof(ExecutionResult),
+                            PROT_READ | PROT_WRITE,
+                            MAP_SHARED | MAP_ANONYMOUS,
+                            -1,
+                            0);
+
+  ExecutionResult *SharedResult = new (SharedMemory) ExecutionResult();
+
+  int ChildPid = fork();
+  if (ChildPid == -1) {
+    exit(1);
+  }
+
+  else if (ChildPid == 0) {
+    Func(SharedResult);
+    exit(0);
+  }
+
+  else {
+    /// TODO: Handle at least 3 cases:
+    /// 1) Child process crashes
+    /// 2) Child process exits 1
+    /// 3) Child process hangs
```

Comment:
We cannot determine if program hangs or not, however this is not the goal.
We just want to stop hanging mutant after some timeout.

C* languages have a nice structure, called for-loop:
```
(for int i = 0; i < 5; i++) {
 /// do stuff
}
```
Here, if we replace `++` with `--` - the program hangs. There is a workaround: to not apply such a mutation in case if it's an index from a for-loop.
However, I imagine many more exceptions like this. It's better to see whether program hangs because of 'wrong' mutation, than to ignore this problem.

We can make many assumptions regarding potential hanging mutants an workaround them in advance, but imo it's better to face them in reality and then solve.

How did you overcome this problem?